jck/build.xml

```diff
@@ -24,6 +24,77 @@
 	<property name="SYSTEMTEST_BUILD_ROOT" value="${BUILD_ROOT}/systemtest" />
 	<property environment="env" />
 
+	<target name="stage_jck_material">
+		<!-- Check if JCK VERSION is correctly set-->
+		<if>
+			<isset property="env.JCK_VERSION" />
+			<then>
+				<echo>=== env.JCK_VERSION is set to ${env.JCK_VERSION} ===</echo>
+			</then>
+			<else>
+				<fail message="env.JCK_VERSION: ${env.JCK_VERSION} was not defined." />
+			</else>
+		</if>
+		<!-- Starting downloading or updating JCK materials based on JCK GIT REPO and JCK VERSION-->
+		<if>
+			<not>
+				<available file="${env.JCK_ROOT}/${env.JCK_VERSION}" type="dir" />
+			</not>
+			<!-- jck materials doesn't exist, download them -->
+			<then>
+				<echo message="${env.JCK_ROOT}/${env.JCK_VERSION} does not exist, 
+					clone from ${env.JCK_GIT_REPO} to ${env.JCK_ROOT}/${env.JCK_VERSION}" />
+				<exec executable="git" dir="${env.JCK_ROOT}" failonerror="true">
+					<arg value="clone" />
+					<arg value="--depth" />
+					<arg value="1" />
+					<arg value="--single-branch" />
+					<arg value="${env.JCK_GIT_REPO}" />
+					<arg value="${env.JCK_VERSION}" />
+				</exec>
+			</then>
+			<!-- jck materials exists, update it materials if needed-->
+			<else>
+				<echo message="${env.JCK_ROOT}/${env.JCK_VERSION} exists, try to update jck materials" />
+				<echo message="git fetch ${env.JCK_GIT_REPO}" />
+				<exec executable="git" dir="${env.JCK_ROOT}/${env.JCK_VERSION}" failonerror="true">
+					<arg value="fetch" />
+				</exec>
+				<echo message="git rebase origin/master" />
+				<exec executable="git" dir="${env.JCK_ROOT}/${env.JCK_VERSION}" failonerror="true">
+					<arg value="rebase" />
+					<arg value="origin/master" />
+				</exec>
+			</else>
+		</if>
+	</target>
+
+	<target name="compile_jck">
+		<propertyregex property="jck_short_version" input="${env.JCK_VERSION}" regexp="jck([^\.]*)" select="\1" casesensitive="false" />
+		<propertyregex property="spec_short_name" defaultValue="${env.SPEC}" input="${env.SPEC}" regexp="([^\.]*)_cmprssptrs" select="\1" casesensitive="false" />
+		<echo message="jck_short_version is ${jck_short_version}" />
+		<echo message="spec_short_name is {spec_short_name}" />
+		<if>
+			<not>
+				<available file="${env.JCK_ROOT}/${env.JCK_VERSION}/natives" type="dir" />
+			</not>
+			<!-- jck natives doesn't exist, compile them -->
+			<then>
+				<exec executable="make" failonerror="true">
+					<arg value="-f" />
+					<arg value="${SYSTEMTEST_ROOT}/openjdk-systemtest/openjdk.test.jck/src/native/makefile" />
+					<arg value="PLATFORM=${spec_short_name}" />
+					<arg value="SRCDIR=${env.JCK_ROOT}/${env.JCK_VERSION}/JCK-runtime-${jck_short_version}" />
+					<arg value="OUTDIR=${env.JCK_ROOT}/${env.JCK_VERSION}/natives" />
+					<arg value="build" />
+				</exec>

```

Comment:
what is the output structure? Will the generated lib be directly placed under `natives` folder? If so, do we need to change `OUTDIR` to be `${env.JCK_ROOT}/${env.JCK_VERSION}/natives/{spec_short_name}`?

Reply:
The compilation will add `${spec_short_name}` automatically

Reply:
also updated check